bson/bsoncodec/string_codec.go

```diff
@@ -23,7 +23,7 @@ type StringCodec struct {
 var (
 	defaultStringCodec = NewStringCodec()
 
-	_ ValueCodec  = defaultStringCodec
+	_ valueCodec  = defaultStringCodec
```

Comment:
Could we add a comment describing the compile check?

Reply:
This compile-time assertion is actually redundant because the codec is already passed into `RegisterTypeEncoder` and `RegisterTypeDecoder`, so I will remove it. However, I will add a comment to the `typeDecoder` assertion below.